### PR TITLE
[FIF-282] Moves React components to EdFi.Buzz.UI - Student Roster page.

### PR DESCRIPTION
**Pull request by DavidJGapCR**
To test this just need to execute the API and the new front end. Attaching some screenshots.

app1

app2

app3

### CODE REVIEW

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+                <div className='bold'>Best time to contact:</div>
+                <div>{student.contacts[0].besttimetocontact}</div>
+              </div>
+            </div>
+          )}
+          {(!student.contacts || student.contacts.length === 0) && (
+            <div className='alert alert-primary'>Student has no contacts</div>
+          )}
+          <div className='outline-button'>
+            <a href={`#/app/studentDetail/${student.studentschoolkey}`}>Student Details</a>
```
Comment (NSD-Xavier) [inline EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx:278]: Use a Link object
```suggestion
            <Link to={`/studentDetail/${student.studentschoolkey}`}>Student Details</Link>
```

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+}
+
+export const StudentCard: React.FunctionComponent<StudentCardComponentProps> = (props: StudentCardComponentProps) => {
+  const {student} = props;
+  const [isCollapsed, setIsCollapsed] = useState(false);
+  return (
+    <StyledStudentCard className='card'>
+      <div className='student-card-body p-t-0'>
+        <div className='d-flex p-t-12'>
+          <a href={`#/app/studentDetail/${student.studentschoolkey}`}>
```
Comment (NSD-Xavier) [inline EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx:187]: Use Link object. Remove #/app.
```suggestion
          <Link to={`/studentDetail/${student.studentschoolkey}`}>
```

**Comment by NSD-Xavier** [comment]
There are some compiling issues.

```
./src/common/Icons.tsx 
 Line 16:57: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 Line 18:57: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 Line 20:56: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 Line 26:63: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 Line 28:61: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 
./src/Feature/StudentRoster/studentCard.tsx 
 Line 189:15: Redundant alt attribute. Screen-readers already announce `img` tags as an image. You don’t need to use the words `image`, `photo,` or `picture` (or any specified custom words) in the alt prop jsx-a11y/img-redundant-alt 
 Line 286:11: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 
./src/Components/SearchInSections/searchInSections.tsx 
 Line 229:11: img elements must have an alt prop, either with meaningful text, or an empty string for decorative images jsx-a11y/alt-text 
 
./src/Feature/StudentRoster/studentTable.tsx 
 Line 28:13: Redundant alt attribute. Screen-readers already announce `img` tags as an image. You don’t need to use the words `image`, `photo,` or `picture` (or any specified custom words) in the alt prop jsx-a11y/img-redundant-alt 
```

**Review by kurtjohnson-HS — CHANGES_REQUESTED** [review]
Looks good to me. I would approve after fixing the linting issues.

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx (hunk 1 above)
Comment (DavidJGapCR) [reply EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx]: Fixed

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx (hunk 2 above)
Comment (DavidJGapCR) [reply EdFi.Buzz.UI/src/Feature/StudentRoster/studentCard.tsx]: Fixed

**Review by kurtjohnson-HS — APPROVED** [review]

**Comment by NSD-Xavier** [comment]
FIF-286 is in main. I have:
- src/assets/fonts
- src/types/font.d.ts 
- globalstyle.ts in src/
- src/globalstyle.ts
- <GlobalFonts /> and <ThemeProvider theme={BuzzTheme}> defined in App.tsx to be global

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          <th style={{ 'marginLeft': '2em' }}>Student Email</th>
+          <th>Primary Contact</th>
+        </tr>
+      </thead>
+      <tbody>
+        {studentList.map(student => <tr key={student.studentschoolkey}>
+          <td>
+            <img src={student.pictureurl} alt='{student.name} Profile' className='image-round'
+              style={{ 'width': '32px' }} />
+            <a href={`#/app/studentDetail/${student.studentschoolkey}`}>
```
Comment (NSD-Xavier) [inline EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx:30]: Use a Link Object

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+          </td>
+          <td>
+            <a href={`mailto:${student.primaryemailaddress}`}>{student.primaryemailaddress}</a>
+          </td>
+
+          {student.contacts && student.contacts?.length > 0 &&
+            <td>
+              {student.contacts[0].contactlastname} {student.contacts[0].contactfirstname}
+                ({student.contacts[0].relationshiptostudent})<br />
+              Preferred Contact Method: {student.contacts[0].relationshiptostudent} <br />
```
Comment (NSD-Xavier) [inline EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx:44]:
```suggestion
              Preferred Contact Method: {student.contacts[0].preferredcontactmethod} <br />
```

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx (hunk 3 above)
Comment (DavidJGapCR) [reply EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx]: Fixed

File: EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx (hunk 4 above)
Comment (DavidJGapCR) [reply EdFi.Buzz.UI/src/Feature/StudentRoster/studentTable.tsx]: Fixed

**Review by NSD-Xavier — APPROVED** [review]

**Pull request merged**